src/Servers/Kestrel/stress/Program.cs

```diff
@@ -0,0 +1,750 @@
+                using (var req = new HttpRequestMessage(HttpMethod.Head, serverUri) { Version = httpVersion })
+                using (HttpResponseMessage m = await ctx.HttpClient.SendAsync(req))
+                {
+                    ValidateResponse(m, httpVersion);
+                    if (m.Content.Headers.ContentLength != maxContentLength)
+                    {
+                        throw new Exception($"Expected {maxContentLength}, got {m.Content.Headers.ContentLength}");
+                    }
+                    string r = await m.Content.ReadAsStringAsync();
+                    if (r.Length > 0) throw new Exception($"Got unexpected response: {r}");
+                }
+            }),
+
+            ("PUT",
+            async ctx =>
+            {
+                string content = ctx.GetRandomSubstring(contentSource);
+                Version httpVersion = ctx.GetRandomVersion(httpVersions);
+
+                using (var req = new HttpRequestMessage(HttpMethod.Put, serverUri) { Version = httpVersion, Content = new StringContent(content) })
+                using (HttpResponseMessage m = await ctx.HttpClient.SendAsync(req))
+                {
+                    ValidateResponse(m, httpVersion);
+                    string r = await m.Content.ReadAsStringAsync();
+                    if (r != "") throw new Exception($"Got unexpected response: {r}");
+                }
+            }),
+        };
+
+        if (listOps)
+        {
+            for (int i = 0; i < clientOperations.Length; i++)
+            {
+                Console.WriteLine($"{i} = {clientOperations[i].Item1}");
+            }
+            return;
+        }
+
+        if (opIndices != null)
+        {
+            clientOperations = opIndices.Select(i => clientOperations[i]).ToArray();
+        }
+
+        Console.WriteLine("     .NET Core: " + Path.GetFileName(Path.GetDirectoryName(typeof(object).Assembly.Location)));
+        Console.WriteLine("  ASP.NET Core: " + Path.GetFileName(Path.GetDirectoryName(typeof(WebHost).Assembly.Location)));
+        Console.WriteLine("       Tracing: " + (logPath == null ? (object)false : logPath.Length == 0 ? (object)true : logPath));
+        Console.WriteLine("   ASP.NET Log: " + aspnetLog);
+        Console.WriteLine("   Concurrency: " + concurrentRequests);
+        Console.WriteLine("Content Length: " + maxContentLength);
+        Console.WriteLine(" HTTP Versions: " + string.Join<Version>(", ", httpVersions));
+        Console.WriteLine("      Lifetime: " + (connectionLifetime.HasValue ? $"{connectionLifetime}ms" : "(infinite)"));
+        Console.WriteLine("    Operations: " + string.Join(", ", clientOperations.Select(o => o.Item1)));
+        Console.WriteLine("   Random Seed: " + seed);
+        Console.WriteLine();
+
+        // Start the Kestrel web server in-proc.
+        Console.WriteLine("Starting server.");
+        WebHost.CreateDefaultBuilder()
```

Comment:
Why isn't the host or builder captured in a variable anywhere? The host needs to be disposed.

Reply:
Convert to generic host.

Reply:
Server setup is a good candidate to extract to its own method.